### PR TITLE
fix(material/tabs): enable hydration

### PR DESCRIPTION
**Pull request by crisbeto**
Adds a workaround that allows us to enable hydration in the tab group component.

### CODE REVIEW

File: src/material/tabs/tab-group.ts [hunk 1]
```diff
@@ -273,6 +277,7 @@ export class MatTabGroup implements AfterContentInit, AfterContentChecked, OnDes
         : false;
     this.stretchTabs =
       defaultConfig && defaultConfig.stretchTabs != null ? defaultConfig.stretchTabs : true;
+    this._isServer = !inject(Platform).isBrowser;
```
Comment (devversion) [inline src/material/tabs/tab-group.ts:280]: Nit: can this be defined directly?

File: tools/public_api_guard/material/tabs.md [hunk 2]
```diff
@@ -319,7 +320,7 @@ export class MatTabGroup implements AfterContentInit, AfterContentChecked, OnDes
...
+    static ɵcmp: i0.ɵɵComponentDeclaration<MatTabGroup, "mat-tab-group", ["matTabGroup"], { "color": { "alias": "color"; "required": false; }; "fitInkBarToContent": { "alias": "fitInkBarToContent"; "required": false; }; "stretchTabs": { "alias": "mat-stretch-tabs"; "required": false; }; "dynamicHeight": { "alias": "dynamicHeight"; "required": false; }; "selectedIndex": { "alias": "selectedIndex"; "required": false; }; "headerPosition": { "alias": "headerPosition"; "required": false; }; "animationDuration": { "alias": "animationDuration"; "required": false; }; "contentTabIndex": { "alias": "contentTabIndex"; "required": false; }; "disablePagination": { "alias": "disablePagination"; "required": false; }; "disableRipple": { "alias": "disableRipple"; "required": false; }; "preserveContent": { "alias": "preserveContent"; "required": false; }; "backgroundColor": { "alias": "backgroundColor"; "required": false; }; }, { "selectedIndexChange": "selectedIndexChange"; "focusChange": "focusChange"; "animationDone": "animationDone"; "selectedTabChange": "selectedTabChange"; }, ["_allTabs"], ["*"], true, never>;
```
Comment (devversion) [inline tools/public_api_guard/material/tabs.md:323]: I have no idea how the diagnostic for the projection with the new control flow works, but does this new "contentSelector" have a potential impact?

File: src/material/tabs/tab-group.html [hunk 3]
```diff
@@ -51,6 +51,16 @@
...
 </mat-tab-header>
 
+<!--
+  We need to project the content somewhere to avoid hydration errors. Some observations:
+  1. This is only necessary on the server.
+  2. We get a hydration error if there aren't any nodes after the `ng-content`.
+  3. We get a hydration error if `ng-content` is wrapped in another element.
+-->
+@if (_isServer) {
+  <ng-content/>
```
Comment (devversion) [inline src/material/tabs/tab-group.html:61]: why is this needed? I don't follow and would like to understand more. Why isn't this needed with CSR?

Comment (devversion) [reply src/material/tabs/tab-group.html:61]: Ohh, I see what you're saying. I would have expected the `mat-tab` to be resolved as well and have it's content SSR-ed as well. That seems like a surprising requirement for projection?

Comment (crisbeto) [reply src/material/tabs/tab-group.html:61]: It is a bit weird, but it's kinda expected. If the `mat-tab` doesn't get projected, when the HTML makes it to the browser, it won't know that it existed when it was created on the server.

Comment (devversion) [reply src/material/tabs/tab-group.html:61]: It feels surprising to me that component authors would need to take care of that. It almost sounds like a bug to me that it works in CSR like that (talking about `ContentChildren` matching without a ng-content`).

The change makes sense, but would love the comment to capture this as well, but no need to re-push. Also cc. @AndrewKushnir as some user-facing feedback on hydration potentially

**Review by devversion — APPROVED** [review]
